modPrecompiled/src/org/aion/precompiled/contracts/ATB/TokenBridgeContract.java

```diff
@@ -250,16 +256,13 @@ public ExecutionResult transfer(@Nonnull final byte[] to,
         if (this.track.getBalance(this.contractAddress).compareTo(value) < 0)
             return new ExecutionResult(ExecutionResult.ResultCode.FAILURE, 0);
 
-        byte[] code = this.track.getCode(new Address(to));
-        if (code != null && code.length != 0)
-            return new ExecutionResult(ExecutionResult.ResultCode.FAILURE, 0);
-
         // otherwise prepare for a transfer
         ExecutionContext innerContext = assembleContext(to, value);
-        IRepositoryCache cache = this.track;
+
         AionInternalTx tx = newInternalTx(innerContext);
```

Comment:
I think we should construct the internal transaction based on the current context.

Reply:
Can you describe why this is necessary? From my POV the execution context is not "left" anywhere in the state

Reply:
Okay, in conclusion: the main deviation from ``Calback`` is that we would have incremented our depth here, we will resolve it by not doing so.

Reply:
Or we can make the `Callback` to start depth at `1`, as the depth of internal transaction is not properly defined. It can be either the depth at which we're creating the internal transaction, or the depth at which the internal transaction is executed. :)